### DISCUSSION TITLE
How to use keygen releases

### DISCUSSION THREAD

**Discussion by MblRobotics** [keygen-sh/community#110] [General]
Hello 
I'm a developer within a company , we're building a python application and now we're in licensing phase 
We decided to go with keygen, currently we create licenses and we give the clients the license key and they just have to enter it in a window dedicated for activation to use the App
Currently , licensing is working well , but we want to use the release feature that sounds interresting
For now , i can create release , upload the artifact and associate it to the release , what i can't understand now how the release can be used , in other words how we can detect which release the running application is using (i mean keygen release object that have an id ... ) and how for example when i publish a new release how the user will be notified , if i want to yank some release so it's users can't use it anymore and have to update the app , is this possible using keygen Release or not?
In resume , i want to understand the release workflow and the best practices when dealing with releases 
Licenses doesn't have a relationship with releases , i can use metadata to save a number indicating the release number but i feel like this is a hack not the best way to do it 
Any help will be appreciated , sorry for the long question

**Comment by ezekg** [comment]
What you're looking for is the [`scope.version`](https://keygen.sh/docs/api/licenses/#licenses-validate-meta-scope.version) parameter for license validations, which sets the license's [`version`](https://keygen.sh/docs/api/licenses/#licenses-object-attrs-version) attribute. This will allow you to track a license's current version.

Regarding notifications, you can set up a [webhook](https://keygen.sh/docs/using-webhooks/), listening for the `release.published` and `release.yanked` webhook events, then using that to trigger e.g. an email notification. Alternatively, for a no-code solution, you can use [Zapier](https://zapier.com/apps/keygen/integrations) for this as well.

**Reply by MblRobotics** [reply]
Thank you ezekg , that was helpful 
Just one another question , i created 3 releaes and published one, yanked one and kept one as DRAFT
when i validate the license and i use version scope i get version_scope_mismatch when the version is not an existing release (which is the expected behaviour) but when i use a version of a draft or yanked release it shows license valid , isn't supposed to validate it only when release is published ? 
Thank you again for yourgreat answer and support

**Reply by ezekg** [reply]
@MblRobotics this should now be fixed.

**Comment by MblRobotics** [comment]
yes , thank you
